mantidproject/mantid
# New algorithm: LineProfile

This PR introduces a new algorithm LineProfile. The algorithm gives a horizontal or vertical cut of a Matrix workspace by averaging or summing the bins over a given line width.

**To test:**

Try giving the algorithm both correct and incorrect input. Does it behave as expected in the 'usual' cases? How about corner cases? Are error/warning messages comprehensible?

A close-to real-life use case is shown below. The *SofQWNormalisedPolygon* algorithm produces a workspace where NaNs signify missing data which should be ignored in the profile.
```python
# Add correct path-to-build below!
buildDir = <path to Mantid's build dir>
Load(
    Filename=dataDir + 'ExternalData/Testing/Data/UnitTest/ILL/IN4/084447.nxs', 
    OutputWorkspace='084447')
ExtractMonitors(
    InputWorkspace='084447',
    DetectorWorkspace='dets', MonitorWorkspace='mons')
CorrectTOFAxis(
    InputWorkspace='dets',
    OutputWorkspace='in_tof',
    IndexType='Workspace Index',
    ElasticBinIndex=175,
    L2=2)
ConvertUnits(
    InputWorkspace='in_tof',
    OutputWorkspace='in_DeltaE',
    Target='DeltaE',
    EMode='Direct',
    ConvertFromPointData=False)
Rebin(
    InputWorkspace='in_DeltaE',
   OutputWorkspace='rebinned_DeltaE',
   Params='-80,1,8',
   PreserveEvents=False)
ws = SofQWNormalisedPolygon(
    InputWorkspace='rebinned_DeltaE',
    OutputWorkspace='SofQW',
    QAxisBinning='0,0.05,8',
    EMode='Direct')
profile = LineProfile(
    InputWorkspace=ws,
    Direction='Horizontal',
    Centre=2,
    HalfWidth=1,
    Start=-50,
    End=-5,
    IgnoreNans=True)
# Comparion to a case where NaNs are replaced by zeros.
ReplaceSpecialValues(
    InputWorkspace=ws,
    OutputWorkspace='infsremoved'
    NanValue=0,
    NanError=0)
profile_NaNs = LineProfile(
    InputWorkspace='infsremoved',
    Direction='Horizontal',
    Centre=2,
    HalfWidth=1,
    Start=-50,
    End=-5,
    IgnoreNans=True)
```

Fixes #19286.

*Release notes were updated accordingly.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
